sys/arduino/base.cpp

```diff
@@ -58,3 +61,44 @@ void delay(unsigned long msec)
 {
     xtimer_usleep(1000 * msec);
 }
+
+#ifdef ADC_NUMOF
+/*
+ * Bitfield for the state of the ADC-channels.
+ * 0: Not initialized
+ * 1: Successfully initialized
+ */
+ static uint16_t adc_state;
+
+int analogRead(int arduino_pin)
+{
+    int adc_value;
+
+    /* Map the arduino pin number to the ADC line */
+    uint16_t adc_line_ = (uint16_t) (arduino_pin - ADC_PIN_OFFSET);
```

Comment:
so given that `ardunio_pin` is a number from `[0:x]` with `x` being the available analog ports on that board, this line is broken when substracting `ADC_PIN_OFFSET` from such a value...